pre_processing/setup.F90

```diff
@@ -1428,16 +1432,21 @@ subroutine setup_slstr(args, source_attributes, channel_ids_user, &
    args%cday = trim(adjustl(l1b_start(index2+8:index2+9)))
    args%chour = trim(adjustl(l1b_start(index2+11:index2+12)))
    args%cminute = trim(adjustl(l1b_start(index2+14:index2+15)))
-   args%csecond = trim(adjustl(l1b_start(index2+17:index2+18)))
+   args%csecond = trim(adjustl(l1b_start(index2+17:index2+25)))
 
    ! get year, doy, hour and minute as integers
    read(args%cyear, '(I4)') args%year
    read(args%cmonth, '(I2)') args%month
    read(args%cday, '(I2)') args%day
    read(args%chour, '(I2)') args%hour
    read(args%cminute, '(I2)') args%minute
-   read(args%csecond, '(I2)') args%second
-   if (args%second .ge. 30) then
+   ! SLSTR file times are rounded to the nearest second, whereas the attributes
+   ! are specified to a micro-second.
+   read(args%csecond(1:len_trim(args%csecond)), '(F9.6)') fsecond
+   args%second = NINT(fsecond)
+   ! Deal with the case where rounding the second carries up to a new minute
+   if (args%second .eq. 60) then
```

Comment:
Not sure what you mean.
Previously the code rounded the _minute_ if the second field (which was read as an integer - presumably this would truncate the fractional second, not round it) was `arg%second .ge. 30`. I'll admit that I'm not 100% whether the time-stamp in a L1B filename is _rounded_ to the nearest second or _rounded down_ to the previous second. I suspect the former though.

Reply:
This no longer rounds to the nearest minute as the outermost if statement `if second == 60`

Reply:
Off the top of my head, we want to read the seconds as a float, then round the seconds to the nearest whole number, then do the previous code block to round to the nearest minute. But that probably repeats a block of code

Reply:
Exactly - it rounds to the nearest second with the `NINT` statement, and then alters the minutes (and hours and days if necessary) if seconds ends up being 60.
I'm not quite sure what your original question was (?)

Reply:
Yes, but I thought the problem was that we wanted to round to the nearest minute and we were using the wrong value of seconds? If the problem was actually that SLSTR doesn't round to the nearest minute then you're fine